src/com/goide/actions/GoCreateFileAction.java

```diff
@@ -41,7 +46,12 @@
   protected PsiFile createFile(String name, @NotNull String templateName, @NotNull PsiDirectory dir) {
     FileTemplate template = FileTemplateManager.getInstance(dir.getProject()).getInternalTemplate(templateName);
     Properties properties = new Properties();
-    properties.setProperty(PACKAGE, ContainerUtil.getLastItem(StringUtil.split(dir.getName(), "-")));
+    Collection<String> packages = GoUtil.getAllPackagesInDirectory(dir);
+    String packageName = packages.isEmpty() ? GoPsiImplUtil.makePackageNameValid(ContainerUtil.getLastItem(StringUtil.split(dir.getName(), "-"))) : ArrayUtil.toStringArray(packages)[0];
```

Comment:
Why do you need to split directory name? Doesn't makePackageNameValid do you for you?